src/Core/Navigation/INavigationService.cs

```diff
@@ -20,63 +20,68 @@ public interface INavigationService
         /// <summary>
         /// Removes an existing mapping between a ViewModel and a View
         /// </summary>
-        /// <typeparam name="TViewModelInterface"></typeparam>
-        void Unregister<TViewModelInterface>()
-            where TViewModelInterface : class, IViewModel;
+        /// <typeparam name="TViewModel"></typeparam>
+        void Unregister<TViewModel>()
+            where TViewModel : class, IViewModel;
 
 
+        /// <summary>
+        /// Removes all existing mappings between ViewModel and View types
+        /// </summary>
+        void ResetRegistrations();
+
 
         /// <summary>
         /// Sets the application's root view using the ViewModel interface and the registered View type
         /// </summary>
-        /// <typeparam name="TViewModelInterface"></typeparam>
+        /// <typeparam name="TViewModel"></typeparam>
         /// <param name="dataContext">A ViewModel instance of the given interface type</param>
-        Task SetRootAsync<TViewModelInterface>()
-            where TViewModelInterface : class, IViewModel;
+        Task SetRootAsync<TViewModel>()
+            where TViewModel : class, IViewModel;
 
 
         /// <summary>
         /// Creates a new instance of the ViewModel interface type, looks up the associated view and displays it
         /// </summary>
-        Task ShowAsync<TViewModelInterface>()
-            where TViewModelInterface : class, IViewModel;
+        Task ShowAsync<TViewModel>()
+            where TViewModel : class, IViewModel;
 
         /// <summary>
         /// Creates a new instance of the ViewModel interface type, initializes it using the parameter, looks up the associated view and displays it
         /// </summary>
-        /// <param name="modelToPass">The data that should be used to initialize the ViewModel</param>
-        Task ShowAsync<TViewModelInterface, TModel>(
-            TModel modelToPass)
-            where TViewModelInterface : class, IViewModel<TModel>;
+        /// <param name="data">The data that should be used to initialize the ViewModel</param>
+        Task ShowAsync<Interface, TData>(
+            TData data)
+            where Interface : class, IViewModel<TData>;
 
 
 
         /// <summary>
         /// Closes the View that is associated with the ViewModel interface type
         /// </summary>
-        Task CloseAsync<TViewModelInterface>()
-            where TViewModelInterface : class, IViewModel;
+        Task CloseAsync<TViewModel>()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to have parameters matching all the type parameters.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)